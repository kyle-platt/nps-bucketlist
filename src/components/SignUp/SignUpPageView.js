import React from 'react';
import { Link } from 'react-router-dom';
import * as ROUTES from '../../constants/routes';

const SignUpPageView = ({
    email,
    error,
    handleChange,
    handleSubmit,
    isInvalid,
    passwordOne,
    passwordTwo,
}) => (
    <div className="app-background">
        <Link to={ROUTES.LANDING}>back</Link>
        <h1>Let's get you signed up</h1>
        <form onSubmit={handleSubmit}>
            <input
                name="email"
                value={email}
                onChange={handleChange}
                type="text"
                placeholder="Email Address"
            />
            <input
                name="passwordOne"
                value={passwordOne}
                onChange={handleChange}
                type="password"
                placeholder="Password"
            />
            <input
                name="passwordTwo"
                value={passwordTwo}
                onChange={handleChange}
                type="password"
                placeholder="Confirm Password"
            />
            <button disabled={isInvalid} type="submit">
                Sign Up
            </button>
            {
                error && (
                    <p>
                        {error.message}
                    </p>
                )
            }
        </form>
        <p>
            Already have an account? <Link to={ROUTES.SIGN_IN}>Sign In</Link>
        </p>
    </div>
);

export default SignUpPageView;
