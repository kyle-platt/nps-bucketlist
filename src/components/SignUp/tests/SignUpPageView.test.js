import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SignUpPageView from '../SignUpPageView';

describe('SignInPageView', () => {
    const defaultProps = {
        email: 'info@test.com',
        error: null,
        handleChange: () => { },
        handleSubmit: () => { },
        isInvalid: false,
        passwordOne: '123abc',
        passwordTwo: '123abc',
    };

    it('renders with some data', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <SignUpPageView
                    {...defaultProps}
                />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
  
    it('renders with an error', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <SignUpPageView
                    {...defaultProps}
                    error={{ message: 'some error' }}
                />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
