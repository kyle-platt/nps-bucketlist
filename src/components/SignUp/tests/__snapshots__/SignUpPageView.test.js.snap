// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPageView renders with an error 1`] = `
<div
  className="app-background"
>
  <a
    href="/"
    onClick={[Function]}
  >
    back
  </a>
  <h1>
    Let's get you signed up
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="text"
      value="info@test.com"
    />
    <input
      name="passwordOne"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123abc"
    />
    <input
      name="passwordTwo"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value="123abc"
    />
    <button
      disabled={false}
      type="submit"
    >
      Sign Up
    </button>
    <p>
      some error
    </p>
  </form>
  <p>
    Already have an account? 
    <a
      href="/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
  </p>
</div>
`;

exports[`SignInPageView renders with some data 1`] = `
<div
  className="app-background"
>
  <a
    href="/"
    onClick={[Function]}
  >
    back
  </a>
  <h1>
    Let's get you signed up
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="text"
      value="info@test.com"
    />
    <input
      name="passwordOne"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123abc"
    />
    <input
      name="passwordTwo"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value="123abc"
    />
    <button
      disabled={false}
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <p>
    Already have an account? 
    <a
      href="/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
  </p>
</div>
`;
